### PR TITLE
Postgresql90: fix build errors

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Updated revision, since changes are generally relevant.

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6 PPC (10A190)
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by barracuda156** [comment]
Hmm, buildbot fail on `configure: error: Cannot find a working 64-bit integer type`. I did not change anything related to 64-bit integer types. Any idea what goes wrong?

**Comment by ryandesign** [comment]
Be aware that there are many postgresql* ports, for the various different postgresql versions, and that it is desirable for them to remain as similar as possible (although other committers before you may not have kept this in mind). For example, whitespace changes or build fixes made to one port should be made equivalently to all the others as needed. (But of course a build fix for one version might not necessarily be needed for all versions.) It is desirable, for example, to be able to look at a diff between the postgresql90 and postgresql91 ports and see only the differences that are relevant between those two versions, and not also see whitespace changes.

**Comment by mascguy** [comment]
> @ryandesign I am not gonna insist on fixing spaces etc. Do you recommend reverting modeline commit? I do not have time to fix every single postgresql version at the moment.

While the cleanup is great, I agree with Ryan that we should do so across all or none.

So go ahead and drop that commit, and we'll merge.

**Review by mascguy — APPROVED** [review]
LGTM. Once CI passes, we'll merge.

**Comment by barracuda156** [comment]
@mascguy @ryandesign Done.

**Comment by barracuda156** [comment]
> LGTM. Once CI passes, we'll merge.

@mascguy The bot still fails on `configure: error: Cannot find a working 64-bit integer type`. I guess it is just broken for newer systems and likely never built for them.

**Comment by mascguy** [comment]
> @mascguy The bot still fails on `configure: error: Cannot find a working 64-bit integer type`. I guess it is just broken for newer systems and likely never built for them.

Actually it looks like this port is broken across-the-board, due to an SSL issue:

```
checking for SSL_library_init in -lssl... no
configure: error: library 'ssl' is required for OpenSSL
```

https://ports.macports.org/port/postgresql90/details/

Do your changes address that too?

**Comment by barracuda156** [comment]
> > @mascguy The bot still fails on `configure: error: Cannot find a working 64-bit integer type`. I guess it is just broken for newer systems and likely never built for them.
> 
> Actually it looks like this port is broken across-the-board, due to an SSL issue:
> 
> ```
> checking for SSL_library_init in -lssl... no
> configure: error: library 'ssl' is required for OpenSSL
> ```
> 
> https://ports.macports.org/port/postgresql90/details/
> 
> Do your changes address that too?

Yes, `openssl 1.0` PG should take care of that.

**Comment by mascguy** [comment]
> @mascguy The bot still fails on `configure: error: Cannot find a working 64-bit integer type`. I guess it is just broken for newer systems and likely never built for them.

Finally tested this locally on 10.15, and sure enough, I'm seeing the same configure error.

Haven't dug into it yet, but it looks like the following test is failing:

```
| typedef long long int ac_int64;
| 
| /*
|  * These are globals to discourage the compiler from folding all the
|  * arithmetic tests down to compile-time constants.
|  */
| ac_int64 a = 20000001;
| ac_int64 b = 40000005;
| 
| int does_int64_work()
| {
|   ac_int64 c,d;
| 
|   if (sizeof(ac_int64) != 8)
|     return 0;                 /* definitely not the right size */
| 
|   /* Do perfunctory checks to see if 64-bit arithmetic seems to work */
|   c = a * b;
|   d = (c + b) / b;
|   if (d != a+1)
|     return 0;
|   return 1;
| }
| main() {
|   exit(! does_int64_work());
| }
configure:24029: result: no
configure:24042: error: Cannot find a working 64-bit integer type.
```

Perhaps the semantics surrounding a `long long int` have changed in later Clang releases, allowing this test to succeed with earlier Clang versions?

@kencu your thoughts?

**Comment by kencu** [comment]
might just need the <stdlib.h> header added?

```
 % clang inttest.c
inttest.c:24:2: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]
 main() {
 ^
inttest.c:25:4: error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]
   exit(! does_int64_work());
   ^
inttest.c:25:4: note: include the header <stdlib.h> or explicitly provide a declaration for 'exit'
1 warning and 1 error generated.
```

**Comment by barracuda156** [comment]
@mascguy @kencu Could someone please add that? I am away from my hardware until the end of month, so can’t do much atm.

**Comment by kencu** [comment]
in the current postgresql source code they just removed the call to "exit".

```
/*
 * These are globals to discourage the compiler from folding all the
 * arithmetic tests down to compile-time constants.
 */
ac_int64 a = 20000001;
ac_int64 b = 40000005;
int does_int64_work()
{
  ac_int64 c,d;
  if (sizeof(ac_int64) != 8)
    return 0;			/* definitely not the right size */
  /* Do perfunctory checks to see if 64-bit arithmetic seems to work */
  c = a * b;
  d = (c + b) / b;
  if (d != a+1)
    return 0;
  return 1;
}
int
main() {
  return (! does_int64_work());
}
```

**Comment by kencu** [comment]
here's the commit, from 6 years ago

https://github.com/postgres/postgres/commit/1c0cf52b39ca3a9a79661129cff918dc000a55eb

**Comment by barracuda156** [comment]
> here's the commit, from 6 years ago
> 
> [postgres/postgres@1c0cf52](https://github.com/postgres/postgres/commit/1c0cf52b39ca3a9a79661129cff918dc000a55eb)
> 
> looks like there are a few places to change it needed

@kencu I will deal with this once I am back home. Thank you.

@mascguy It may not hurt merging this PR first, IMO, since it _does_ fix `postgresql90` build for some systems, while not for all, as yet.

**Comment by reneeotten** [comment]
@barracuda156 are you planning to finish this up - if not please close the PR

**Comment by barracuda156** [comment]
> @barracuda156 are you planning to finish this up - if not please close the PR

I will finish. Gonna deal with most of the stuff by end of month. Noting is forgotten, things just take time, and right now free time is scarce.

**Comment by barracuda156** [comment]
@reneeotten Are we good to merge this?

**Pull request merged**
